azure_usage/src_notebook/analysis.py

```diff
@@ -215,25 +248,64 @@ def analyse_spnsr_usage(data_path, date_from, date_to, spnsr_budget):
     spnsr_usage_ym_df = group_year_month(sub_data_df)
 
     # adding zeros for the missing year-months
-    spnsr_usage_ym_df = add_missing_year_months(spnsr_usage_ym_df,
-        date_from, date_to, CONST_COL_NAME_COST)
+    spnsr_usage_ym_df = add_missing_year_months(
+        spnsr_usage_ym_df, date_from, date_to, CONST_COL_NAME_COST
+    )
 
     # estimates ideal and available costs
-    spnsr_ideal_avail_ym_df, max_reg_cost_day = get_ideal_avail_spnsr_costs(sub_data_df,
-        date_from, date_to, spnsr_budget)
+    spnsr_ideal_avail_ym_df, max_reg_cost_day = get_ideal_avail_spnsr_costs(
+        sub_data_df, date_from, date_to, spnsr_budget
+    )
 
     # reset index
     spnsr_usage_ym_df = spnsr_usage_ym_df.reset_index(drop=True)
     spnsr_ideal_avail_ym_df = spnsr_ideal_avail_ym_df.reset_index(drop=True)
 
     # combining actual, ideal and available costs into one dataframe
-    spnsr_usage_df = pd.concat([spnsr_ideal_avail_ym_df,
-        spnsr_usage_ym_df.drop([CONST_COL_NAME_YM], axis=1)], axis=1)
+    spnsr_usage_df = pd.concat(
+        [spnsr_ideal_avail_ym_df, spnsr_usage_ym_df.drop([CONST_COL_NAME_YM], axis=1)],
+        axis=1,
+    )
 
     return spnsr_remain, spnsr_usage, spnsr_usage_df, max_reg_cost_day
 
-def get_ideal_avail_spnsr_costs(sub_raw_data_df, spnsr_date_from, spnsr_date_to,
-    spnsr_budget, use_days = False):
+
+def analyse_spnsr_usage_by_service(data_path, date_from, date_to):
```

Comment:
New function

Reply:
Could you please write unittests for the new function.